--- a/drivers/gpu/drm/rockchip/rockchip_vop_reg.c	2017-03-20 08:32:54.643609331 +0100
+++ b/drivers/gpu/drm/rockchip/rockchip_vop_reg.c	2017-03-20 08:33:55.524534131 +0100
@@ -234,9 +234,9 @@
  */
 static const struct vop_win_data rk3288_vop_win_data[] = {
 	{ .base = 0x00, .phy = &rk3288_win01_data,
-	  .type = DRM_PLANE_TYPE_PRIMARY },
+	  .type = DRM_PLANE_TYPE_OVERLAY },
 	{ .base = 0x40, .phy = &rk3288_win01_data,
-	  .type = DRM_PLANE_TYPE_OVERLAY },
+	  .type = DRM_PLANE_TYPE_PRIMARY },
 	{ .base = 0x00, .phy = &rk3288_win23_data,
 	  .type = DRM_PLANE_TYPE_OVERLAY,
 	  .area = rk3288_area_data,
